### PR TITLE
[xalan-c] linux support, fix #6481

### LINKED ISSUE #6481: xalan-c install fails on linux
Package install fails (I did run update but that did not help).

./vcpkg install xalan-c
The following packages will be built and installed:
    xalan-c[core]:x64-linux
Starting package 1/1: xalan-c:x64-linux
Building package xalan-c[core]:x64-linux...
-- Note: xalan-c only supports dynamic library linkage. Building dynamic library.
-- Using cached /home/projects/meme/vcpkg/downloads/xalan_c-1.11-src.zip
-- Using cached /home/projects/meme/vcpkg/downloads/0008-remove-unary-binary-function.patch
-- Using cached /home/projects/meme/vcpkg/downloads/0009-remove-select-workaround.patch
-- Using source at /home/projects/meme/vcpkg/buildtrees/xalan-c/src/1.11-7013b9023f
-- Building c/projects/Win32/VC10/AllInOne/AllInOne.vcxproj for Release
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:56 (message):
    Command failed: msbuild;/home/projects/meme/vcpkg/buildtrees/xalan-c/x64-linux-rel/1.11-7013b9023f/c/projects/Win32/VC10/AllInOne/AllInOne.vcxproj;/p:Configuration=Release;/t:Rebuild;/p:Platform=x64;/p:PlatformToolset=external;/p:VCPkgLocalAppDataDisabled=true;/p:UseIntelMKL=No;/p:WindowsTargetPlatformVersion=;/m;/p:XERCESCROOT=/home/projects/meme/vcpkg/installed/x64-linux
    Working Directory: /home/projects/meme/vcpkg/buildtrees/xalan-c/x64-linux-rel/1.11-7013b9023f
    Error code: No such file or directory
    See logs for more information:

Call Stack (most recent call first):
  scripts/cmake/vcpkg_install_msbuild.cmake:157 (vcpkg_execute_required_process)
  ports/xalan-c/portfile.cmake:57 (vcpkg_install_msbuild)
  scripts/ports.cmake:71 (include)

Error: Building package xalan-c:x64-linux failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: xalan-c:x64-linux
  Vcpkg version: 2018.11.23-unknownhash

### PR DESCRIPTION
**Pull request by isanych**
fix #6481

### CODE REVIEW

**Comment by crusader-mike** [comment]
@ras0219-msft, apologies for bothering you... How soon do you plan to merge this PR? (I am kind of looking forward to use vckg-packaged `xalan-c` in my Linux build)

File: ports/xalan-c/portfile.cmake [hunk 1]
```diff
@@ -46,36 +46,87 @@ vcpkg_extract_source_archive_ex(
...
-    LICENSE_SUBPATH c/LICENSE
-    SKIP_CLEAN
-)
+    set(XALANCROOT ${SOURCE_PATH}/c)
+    set(ENV{XALANCROOT} ${XALANCROOT})
+    foreach(BUILD_TYPE IN LISTS BUILD_TYPES)
+        if(BUILD_TYPE STREQUAL "release")
+            set(P ${CURRENT_PACKAGES_DIR})
+            set(SHORT rel)
+            set(OPTS)
```
Comment (vicroms) [inline ports/xalan-c/portfile.cmake]: Is there a reason why the includes `-I${CURRENT_INSTALLED_DIR}/include` are not passed to the release build?

File: ports/xalan-c/portfile.cmake [hunk 2]
```diff
@@ -46,36 +46,87 @@ vcpkg_extract_source_archive_ex(
...
+        file(REMOVE_RECURSE ${DIR})
+        file(MAKE_DIRECTORY ${DIR})
+        message(STATUS "Configuring ${TARGET_TRIPLET}-${SHORT}")
+        vcpkg_execute_required_process(
+            COMMAND ${XALANCROOT}/runConfigure -p linux -c gcc -x g++ -b ${BITS} -P ${P} ${OPTS}
+            WORKING_DIRECTORY ${DIR}
+            LOGNAME configure-${TARGET_TRIPLET}-${SHORT}
+        )
+        message(STATUS "Building ${TARGET_TRIPLET}-${SHORT}")
+        vcpkg_execute_required_process(
```
Comment (vicroms) [inline ports/xalan-c/portfile.cmake]: This should use `vcpkg_execute_build_process()` instead.

**Comment by vicroms** [comment]
Thanks for the PR!

**Comment by crusader-mike** [comment]
@isanych, `./vcpkg install xalan-c` fails on Linux -- looks like `MsgCreator` fails to build due to missing curl dependencies. I am not sure how to hook it up to libcurl provided by `vcpkg`.

Another this I noticed is this:
```
$ ls -l installed/x64-linux/lib
-rw-r--r--. 1 user user 12924800 Jul 29 19:24 installed/x64-linux/lib/libxerces-c-3.2.a
lrwxrwxrwx. 1 user user       17 Jul 29 19:24 installed/x64-linux/lib/libxerces-c.so -> libxerces-c-3.2.a
```
This looks very weird to me... Any ideas what to do with this?

**Pull request merged**